ledger/ledgercore/statedelta.go

```diff
@@ -149,3 +152,31 @@ func (ad *AccountDeltas) upsert(br basics.BalanceRecord) {
 	}
 	ad.acctsCache[addr] = last
 }
+
+func (sd *StateDelta) Compress() {
```

Comment:
nit: don't use the word "Compress" since we don't really compress anything here. Maybe OptimizeAllocatedMemory ?